src/Event.tsx

```diff
@@ -28,20 +28,37 @@ interface EventProps {
 }
 
 interface EventState {
-  event?: core.Event
+  event?: core.Event,
+  grayedOut: boolean
 }
 
 // TODO: Need to make this component auto-refresh
 class Event extends React.Component<EventProps, EventState> {
   constructor(props: EventProps) {
     super(props)
-    this.state = {}
+    this.state = {
+      grayedOut: true
+    }
   }
 
   async componentDidMount(): Promise<void> {
+    const eventData = await getClient().core().events().get(this.props.id)
+    const projectId = eventData.projectID
+
     this.setState({
-      event: await getClient().core().events().get(this.props.id)
+      event: eventData
     })
+
+    if(projectId) {
+      const authzList = await getClient().core().projects().authz().roleAssignments().list(projectId)
+      const userId = await getUserID()
+
+      for(const authzItem of authzList.items) {
+        if(authzItem.role === "PROJECT_USER" && authzItem?.principal.id === userId) {
+          this.setState({grayedOut: false})
+        }
+      }
```

Comment:
for(const authzItem of authzList.items) {
  if(authzItem.role === "PROJECT_USER") {
    this.setState({grayedOut: false})
    break
  }
}

Reply:
Also I thought for sure that the SDK had constants to represent different roles, but it doesn't. I'll fix that for the next release of the SDK.

Reply:
Done ! Will be there in the next commit.